backend/src/errors.rs

```diff
@@ -0,0 +1,13 @@
+error_chain! {
+    links {
+    }
+
+    foreign_links {
+        ::std::io::Error, Io, "io error";
+        ::json::Error, Json, "json error";
+        ::chrono::ParseError, Chrono, "chrono parsing error";
```

Comment:
I think something simpler should work; maybe `quick_error!` or a hand-crafted basic implementation of the `From<OtherError> -> MyError` logic. I used `error-chain` in part because I sort of expected there to be more here (and for the `errors` declaration below to fill out, as per my comment); and I thought that error-chain seemed to be the way of doing that. I don't feel strongly either way about it; but I feel like the ergonomics of being able to use try! on pretty much any crate's error/result type are nice.